app/src/ui/app.tsx

```diff
@@ -1415,22 +1419,21 @@ export class App extends React.Component<IAppProps, IAppState> {
       this.state.currentFoldout &&
       this.state.currentFoldout.type === FoldoutType.AppMenu
 
-    // As Linux still uses the classic Electron menu, we are opting out of the
-    // custom menu that is shown as part of the title bar below
-    if (__LINUX__) {
+    if (__LINUX__ && this.props.appStore.getState().titleBarStyle == 'native') {
       return null
     }
 
     // When we're in full-screen mode on Windows we only need to render
     // the title bar when the menu bar is active. On other platforms we
     // never render the title bar while in full-screen mode.
+
     if (inFullScreen) {
-      if (!__WIN32__ || !menuBarActive) {
+      if (!menuBarActive) {
         return null
       }
     }
 
-    const showAppIcon = __WIN32__ && !this.state.showWelcomeFlow
```

Comment:
🤔 what's the context on dropping these `__WIN32__` mentions here?

Reply:
That would be unnecessary since we already checked OS version a few lines earlier. (line 1422)